[New Module] Sharpcradle - Execute .NET assembly from URL in memory

hackabean opened the pull request:
- Added sharpcradle

Description and screenshoot:

![image](https://user-images.githubusercontent.com/50614334/66004772-20a00d00-e499-11e9-9b26-3efc0311e5e4.png)

I have noticed that assemblies over 600kb+ tend to crash the stager. Happened to me with a SharpView few times. 

Also finding a reliable way to parse the arguments might be important as some of those tools require quite a range of syntax(es) that ST might struggle to comprehend.

byt3bl33d3r commented:
@hackabean  so since this is essentially 2/4 of what the `execute-assembly` module does, wouldn't be better if we just added a URL parameter to that module instead of a completely new one?

hackabean commented:
Oh yeah, I am happy to merge it with your module and add URL as an option.

byt3bl33d3r commented:
@hackabean yeah that's perfect, that would be awesome. Thanks!

hackabean commented:
@byt3bl33d3r 

- Execute-assembly V2 
Intro:
![image](https://user-images.githubusercontent.com/50614334/66091875-1ce3b780-e579-11e9-8515-03c02a633c99.png)

1. Added execute local-assembly (original execute-assembly with minor changes - fixes #104)

![image](https://user-images.githubusercontent.com/50614334/66092064-f70ae280-e579-11e9-92a7-e2b4a86abc37.png)


2. Added execute remote-assembly (known as sharpcradle)

![image](https://user-images.githubusercontent.com/50614334/66092158-5f59c400-e57a-11e9-8fde-487e44d21589.png)

Since both of those use the same arguments structure I did not need to add another option.

3. Added execute remote-project to load *.xml and *.csproj files as a bonus.
Stager connection via xml:

![image](https://user-images.githubusercontent.com/50614334/66092227-c2e3f180-e57a-11e9-86d7-ccfc950cd6fc.png)

Does not support arguments as it is a project. Don't know if I need to make it explicit. 

- Amended powershell.boo module to fix #105 (See comments) 

- Added import System.Management.Automation to stage.boo to fix #105

byt3bl33d3r commented on core/teamserver/modules/boo/execute-assembly.py:
```diff
@@ -8,15 +8,26 @@ class STModule(Module):
+            'Local-Assembly': {
+                'Description'   :   'Path to local assembly',
                 'Required'      :   True,
                 'Value'         :   ''
```
@hackabean seems like the tests are failing because there isn't a default value here. Can you take a look at the module tests and set it manually through there? 
Thanks!

hackabean replied:
The only error that appears comes from: 
``
2019-10-04 21:41:20,689 2998 MainThread - [ERROR] loader.py: get_loadables - Failed loading listener core/teamserver/listeners/wmi.py: No module named 'impacket'
``
Which as far as I have tested does not prevent ST from running correctly.

hackabean replied:
@byt3bl33d3r I added default values and still nothing. Teamserver is loading modules without any errors.

byt3bl33d3r replied:
@hackabean no worries, i'll take a look later today.
Cheers

Vokaysh commented on core/teamserver/modules/boo/execute-assembly.py:
```diff
@@ -65,12 +65,12 @@ def payload(self):
     	    with open('core/teamserver/modules/boo/src/remote-assembly.boo', 'r') as module_src:
                 src = module_src.read()
-                src = src.replace('BINARY', str(self.options['RemoteAssemebly']['Value']))
+                src = src.replace('BINARY', str(self.options['Remote-Assemebly']['Value']))
```
Assembly* - got an extraneous "e" there
